### PR TITLE
add dark mode and interaction

### PR DESCRIPTION
**Pull request by tcase360**

### CODE REVIEW

**Review by AetherUnbound — APPROVED** [review]
Definitely wait for @sarayourfriend's review on this, but the preview looks amazing! Great job!

File: main.js
Comment (sarayourfriend) [inline main.js:37]: Nice idea storing it in local storage!

File: main.js
Comment (sarayourfriend) [inline main.js]: Where does `toggleButton` come from?

Comment (tcase360) [reply main.js]: Whoops, was meant to be a function call to the toggle function for the correct keys. Fixed.

File: _includes/style.scss
Comment (sarayourfriend) [inline _includes/style.scss]:
```suggestion
    color: $secondary-color;
```

File: _includes/style.scss
Comment (sarayourfriend) [inline _includes/style.scss]: Should these URLs be absolute, just in case?
```suggestion
            background-image: url('/icons/phone-light.svg');
```

File: main.js
Comment (sarayourfriend) [inline main.js:7]: It looks like `:root` resolves to the `html` element instead of the `body`. I wonder if we should just use `body` everywhere we're using root? The class doesn't seem to be applying correct onto the `html` element.

Comment (sarayourfriend) [reply main.js:7]: Wait never mind I think it's still just the issue where it doesn't work if you start off from light.

File: _includes/style.scss
Comment (tcase360) [reply _includes/style.scss]: This is outdated now, not using urls

File: _includes/resource.html
Comment (sarayourfriend) [inline _includes/resource.html:15]: To avoid repeating the same glyph all over the page I wonder if we could define it once and then use the [`use`](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/use) element to reference it? I believe it'll still be styleable.

I've never tried this before though, but it seems wrong to repeat the glyph over and over again on the page.

Comment (AetherUnbound) [reply _includes/resource.html:15]: Do you think we should just define it once as a hidden SVG and then just reference the definition on each resource? I'm not sure how we'd do it easily otherwise

Comment (sarayourfriend) [reply _includes/resource.html:15]: Yeah that'd be the idea I think.

Comment (sarayourfriend) [reply _includes/resource.html:15]: Although I wonder if we could just have `phone-light` and `phone-dark` and then use CSS selectors to toggle one or the other as `display: none` depending on the selected color scheme? Then we can render both as `img src="/whatever"` instead of inlining the SVG.

Comment (AetherUnbound) [reply _includes/resource.html:15]: Oh that makes sense! I can play around with that - do you mind if I do that in a follow up PR though? I'd love to get this merged & the repo transferred and I don't want it contingent on my ability to mess with the proper display modes :sweat_smile:

Comment (sarayourfriend) [reply _includes/resource.html:15]: Yup sounds good!

**Review by sarayourfriend — APPROVED** [review]
LGTM!

**Comment by AetherUnbound** [comment]
I noticed that the button's borders aren't changing color. Gonna fiddle with that a bit before merging, but if I don't come to a quick solution I'll figure it out in a new branch.

**Comment by AetherUnbound** [comment]
Got it!

image

**Pull request merged**
